components/ieee802154/driver/esp_ieee802154_dev.c

```diff
@@ -803,11 +800,11 @@ static inline esp_err_t ieee802154_transmit_internal(const uint8_t *frame, bool
     IEEE802154_SET_TXRX_PTI(IEEE802154_SCENE_TX);
 
     if (cca) {
-        ieee802154_set_cmd(IEEE802154_CMD_CCA_TX_START);
         ieee802154_set_state(IEEE802154_STATE_TX_CCA);
+        ieee802154_set_cmd(IEEE802154_CMD_CCA_TX_START);
     } else {
-        ieee802154_set_cmd(IEEE802154_CMD_TX_START);
         ieee802154_set_state(IEEE802154_STATE_TX);
+        ieee802154_set_cmd(IEEE802154_CMD_TX_START);
```

Comment:
Due to we have already entering the critical, all isrs related to IEEE802154 are disabled. So it might be no need to consider the order of setting state and start an operation on HW.